infra-examples/aws/k8s-cluster/main.tf

```diff
@@ -0,0 +1,219 @@
+#
+#------------------------------------------------------------------------------
+
+locals {
+  # Used by Karpenter config to determine correct partition (i.e. - `aws`, `aws-gov`, `aws-cn`, etc.)
+  partition = data.aws_partition.current.partition
+
+  tags = {
+    "Name"                          = var.name
+    "openedx-k8s-harmony/name"      = var.name
+    "openedx-k8s-harmony/region"    = var.aws_region
+    "openedx-k8s-harmony/terraform" = "true"
+  }
+
+}
+
+module "eks" {
+  source                          = "terraform-aws-modules/eks/aws"
+  version                         = "~> 19.13"
+  cluster_name                    = var.name
+  cluster_version                 = var.kubernetes_cluster_version
+  cluster_endpoint_private_access = true
+  cluster_endpoint_public_access  = true
+  vpc_id                          = data.aws_vpc.reference.id
+  subnet_ids                      = data.aws_subnets.private.ids
+  create_cloudwatch_log_group     = false
+  enable_irsa                     = true
+
+  # NOTE:
+  # by default Kubernetes secrets are encrypted with this key. Add your IAM
+  # user ARN to the owner list in order to be able to view secrets.
+  # AWS EKS KMS console: https://us-east-2.console.aws.amazon.com/kms/home
+  #
+  # audit your AWS EKS KMS key access by running:
+  # aws kms get-key-policy --key-id ADD-YOUR-KEY-ID-HERE --region us-east-2 --policy-name default --output text
+  create_kms_key = var.eks_create_kms_key
+  kms_key_owners = var.kms_key_owners
+
+  # Add your IAM user ARN to aws_auth_users in order to gain access to the cluster itself.
+  # Note that alternatively, the cluster creator (presumably, you) can edit the manifest
+  # for kube-system/aws-auth configMap, adding additional users and roles as needed. 
+  # see: 
+  manage_aws_auth_configmap = true
+  aws_auth_users            = var.map_users
+
+  tags = merge(
+    local.tags,
+    # Tag node group resources for Karpenter auto-discovery
+    # NOTE - if creating multiple security groups with this module, only tag the
+    # security group that Karpenter should utilize with the following tag
+    { "karpenter.sh/discovery" = var.name }
+  )
+
+  # AWS EKS add-ons that are required in order to support persistent volume 
+  # claims for ElasticSearch and Caddy (if you opt for this rather than nginx).
+  # Other addons are required by Karpenter and other optional supporting services.
+  # 
+  # see: https://docs.aws.amazon.com/eks/latest/userguide/eks-add-ons.html
+  cluster_addons = {
+    # required to support internal networking between containers
+    vpc-cni = {
+      addon_version = "v1.12.6-eksbuild.2"
+    }
+    # required to support internal DNS name resolution within the cluster
+    coredns = {
+      addon_version = "v1.10.1-eksbuild.1"
+    }
+    # required to maintain network rules on  nodes and to enable internal
+    # network communication between pods.
+    kube-proxy = {
+      addon_version = "v1.27.1-eksbuild.1"
```

Comment:
@gabor-boros I was wondering what is the best way to handle the EKS cluster add-ons. When you don't specify a version for them (I'm talking specifically of the 4 add-ons included here), the module installs a "default" version of each addon that is compatible with the K8S version of the cluster (not necessarily the latest one). The only exception is the **aws-ebs-csi-driver** add-on, for this one the latest version available is installed. When updating the K8S version, the module will try to get the proper add-on versions for the target K8S version (thanks to the block [here](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/main.tf#L378-L438) available on 19.x version of the module)

I was considering not adding the versions of the add-ons and relying on the module features; the configuration would be like this:

```
cluster_addons = {
    # required to support internal networking between containers
    vpc-cni = {
      name = "vpc-cni"
    }
    # required to support internal DNS name resolution within the cluster
    coredns = {
      name = "coredns"
    }
    # required to maintain network rules on  nodes and to enable internal
    # network communication between pods.
    kube-proxy = {
      name = "kube-proxy"
    }
    # Required for release 1.22 and newer in order to support persistent volume
    # claims for ElasticSearch and Caddy (if you opt for this rather than nginx).
    aws-ebs-csi-driver = {
      name                     = "aws-ebs-csi-driver"
      service_account_role_arn = aws_iam_role.AmazonEKS_EBS_CSI_DriverRole.arn
    }
  }
```
What do you think about it? cc @lpm0073

Reply:
@jfavellar90 I think that's a great idea! Let's not add the version numbers then. If that ensures we are not need to play puzzle with version number compatibility, I'm even more happy.

Reply:
It was tested and it's working as expected